src/NUnitEngine/nunit.engine/Services/DomainManager.cs

```diff
@@ -166,7 +166,10 @@ public void Unload()
                 }
 
                 if (_unloadException != null)
-                    throw new NUnitEngineException("Exception encountered unloading AppDomain", _unloadException);
+                {
+                    var msg = string.Format("Exception encountered unloading AppDomain '{0}': {1}", _domain.FriendlyName, _unloadException.Message);
+                    throw new NUnitEngineException(msg, _unloadException);
```

Comment:
The problem I see here is that _domain.FriendlyName may throw an exception as well. This will happen, for example, if the original exception is thrown for an appdomain that no longer exists. In fact, if memory serves, that's why the message doesn't include the domain name.

You could probably work around this by changing the code inside the try to fetch the FriendlyName and saving it in a member. Your message would need to be different, depending on whether that member contained a name or not. Or you could initialize it to "UNKNOWN".